controllers/routesController.js

```diff
@@ -0,0 +1,115 @@
+import { route } from '../models';
+import { success, fail, sendError } from '../function/respond.js';
+import {
+	validateRoutesOnCreate,
+} from '../function/validation';
+import { paginate } from '../utils/paginate.js';
+
+const addRoute = async (req, res) => {
+
+    const { name, code, city, startLocation, endLocation, duration, distance } = req.body;
+	const { error } = validateRoutesOnCreate({name: name, code: code, city: city, startLocation: startLocation, endLocation: endLocation, duration: duration, distance: distance })
```

Comment:
It's better to do your validation in the routes as  a middleware rather than in the controller.

Reply:
I created a middleware for route validation.